### DISCUSSION TITLE
relationship between Eb/N0 and the raw BER before decoding.

### DISCUSSION THREAD

**Discussion by WuShengHan** [NVlabs/sionna#640] [Q&A]
Hi, I have a question regarding the LDPC5GDecoder(enc, num_iter=?) function. Specifically, I’d like to know if setting num_iter=0 or num_iter=1 could provide information about the initial error count when the data first enters the decoder. My goal is to understand the relationship between Eb/N0 and the raw BER before decoding.

Would setting num_iter to 0 or 1 allow me to extract the raw bit error rate (BER) at different Eb/N0 levels to analyze the impact of noise and signal attenuation in transmission?

Thanks in advance!

**Comment by SebastianCa** [comment]
Hi @WuShengHan,

If you’re interested in the raw BER before decoding, no decoder is needed. You can simply hard-decide the input LLRs intended for the decoder and then use the [compute_ber](https://nvlabs.github.io/sionna/api/utils.html#sionna.utils.compute_ber) function to calculate the BER.
